src/command.rs

```diff
@@ -558,6 +641,11 @@ impl<'l, const C: usize> TryFrom<&'l Data<C>> for Register<'l> {
         info_now!("parsed secret {:?}", &secret);
 
         let kind: oath::Kind = secret_header[0].try_into()?;
+        if kind == Kind::Hmac {
+            // TODO encode verification logic into separate types
+            ensure(secret.len() == 20, FAILED_PARSING_ERROR)?;
```

Comment:
Please use a named constant rather than a magic number here.

Reply:
Agreed, that magic numbers should not be used in the code.
That's a debug assertion, that I have removed that in the end (I should have use `debug_assert` here probably instead).